cli/lib/exec/xvfb.js

```diff
@@ -41,7 +43,33 @@ module.exports = {
   },
 
   isNeeded () {
-    return os.platform() === 'linux' && !process.env.DISPLAY
+    if (os.platform() !== 'linux') {
+      return false
+    }
+
+    if (process.env.DISPLAY) {
+      const issueUrl = util.getGitHubIssueUrl(4034)
+
+      const message = stripIndent`
+        DISPLAY environment variable is set to ${process.env.DISPLAY} on Linux
```

Comment:
being verbose is good, hopefully, someone reads it and understand the problem, even if we work around it using our XVFB